packages/@lwc/shared/src/meta.ts

```diff
@@ -0,0 +1,8 @@
+/*
+ * Copyright (c) 2018, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: MIT
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/MIT
+ */
+// Increment whenever the LWC template compiler changes
```

Comment:
Do we have any way of enforcing this? Or will it be up to the dev/reviewers to remember to do this?

Reply:
We'll have to just remember to do it, whenever the compiler output changes in a functional way. Not sure if there's a way to automate it.

Reply:
You're effectively defining an [ABI](https://en.wikipedia.org/wiki/Application_binary_interface) for LWC. It should be possible to define that surface area and test it.

Reply:
I agree with @kevinv11n's assertion here. Today we consider that the LWC engine and compiler versions are the same. This is why we only run integration tests for the `@lwc/engine-core`. If we want to go down this path, I think it would make sense to invest in template API unit tests. Whenever, a template API unit test changes, it would mean that we would have to bump the compiler version.

Reply:
We discussed this in Dev Sync and decided to make the following changes:

1. Use the regular `package.json` version instead of a separate versioning scheme
2. Warn, don't throw an error
3. Compare the major/minor

This lowers the risk / ambitiousness of this PR, and should address the concerns above.